codex/marketplace.md

```diff
@@ -0,0 +1,209 @@
+  uint256 expiry;
+
+  // random value to differentiate from other requests
+  byte32 nonce;
+
+}
+
+```
+
+`cid` 
+
+An identifier used to locate the dataset
+- MUST be a sha2-256 hash (length of 32 bytes),
+- MUST be generated by the requester
+
+`reward`
+
+- SHOULD be a token known to the network.
+- it MUST be paid directly to storage providers who fill slots for the `duration` of the request
+
+`collateral`
+
+All storage providers MUST provide token collateral before being able to fill a storage slot.
+The collateral is used to punish a storage provider for being an malicious or unreliable.
+The following is related to storage provider who has offered `collateral`
+
+If a storage provider, filling a slot,
+fails to provide enough proofs of storage, the `collateral` MUST be forfeited.
+The storage provider MAY be able to fill the same failed slot,
+but MUST replace any `collateral` that was already forfeited.
+
+A portion of the `collateral` MUST be offered as a reward to validator nodes,
+and a portion SHOULD be offered as a reward to other storage providers that repair empty [slots](#slots).
+
+`proofProbability`
+
+Storage providers are REQUIRED to provide proofs of storage that are submited to the marketplace smart contract and
+verified by validator nodes.
+The requester SHOULD provide the value for the frequency of proofs provided by storage providers.
+
+`duration`
+
+- Once the `reward` has depleted from periodic storage provider payments,
+the storage request SHOULD end.
+The requester MAY renew the storage request by creating a new request with the same `cid` value.
+- Data MAY be considered lost during contract `duration` if storage providers fail to provide storage proofs and
+no other storage providers decide to fill empty slots, see [slots](#slots) below.
+
+### Fulfilling Requests
+In order a storage request to begin,
+storage providers MUST enter a storage contract with the requester via the marketplace smart contract.
+
+When storage providers are selected to fill a slot for the request,
+storage providers MUST NOT abandon the slot, unless the request is canceled or complete.
+If canceled, the slot state SHOULD be changed to cancelled or failed.
+
+#### Slots
+Slots is a method used by the Codex network to distribute data chucks amongst storage providers.
+Data chucks, created by clients nodes, MUST use a method of distributing the dataset for data resiliency.
+- Client nodes SHOULD decide how many nodes should fill the slots of a storage contract.
+- Storage providers MUST be selected to fill a slot,
+
+Each slot represents a chunk of a dataset provided during storage request.
+The first state of a slot is reserved, meaning that the slot is waiting to be reserved by a storage provider.
+The Codex network selects storage providers by using a variation of the Kademila distance function, 
+as described in [CODEX/NODE-DISPERSAL](https://github.com/vacp2p/rfc-index/blob/codex-disperal/codex/slot-dispersal.md).
```

Comment:
Proposed a separate RFC for dispersal, currently in draft. Should this stay separate or combine into this rfc? https://github.com/vacp2p/rfc-index/pull/39